packages/core-data/src/utils/test/set-nested-value.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import setNestedValue from '../set-nested-value';
+
+describe( 'setNestedValue', () => {
+	it( 'should return the same object unmodified if path is an empty array', () => {
```

Comment:
We should probably add some tests to ensure that it behaves correctly if input is not an object, e.g. `null` or `undefined`. I'll leave it to your judgement what "behaves correctly" should mean; `lodash` simply does nothing, but throwing an error may be a good alternative as well.

Reply:
Looks great, thank you!